### PR TITLE
Sync paket project and add CI support

### PR DESCRIPTION
**Pull request by Frassle**
### Purpose of this PR

* Adds a check to travis/appveyor that paket is in sync with the project files.
* Remove conditional StyleCop support, it didn't play well with paket.
* Ran paket install to update the conditions and references in the project files.

### CODE REVIEW

File: src/Generator.Bind/Generator.Bind.csproj [hunk 1]
```diff
@@ -274,4 +274,12 @@
     <Folder Include="Specifications\Docs\" />
   </ItemGroup>
   <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/Generator.Bind/Generator.Bind.csproj]: Is this not in StyleCop.props?

File: src/Generator.Converter/Generator.Convert.csproj [hunk 2]
```diff
@@ -155,7 +156,14 @@
       </ItemGroup>
     </When>
   </Choose>
-  <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/Generator.Converter/Generator.Convert.csproj]: Same here.

File: src/Generator.Rewrite/Generator.Rewrite.csproj [hunk 3]
```diff
@@ -201,7 +202,14 @@
       </ItemGroup>
     </When>
   </Choose>
-  <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/Generator.Rewrite/Generator.Rewrite.csproj]: And here.

File: src/OpenTK.GLWidget/OpenTK.GLWidget.csproj [hunk 4]
```diff
@@ -161,5 +162,12 @@
     </When>
   </Choose>
   <Import Project="..\..\packages\gtk-sharp3\build\gtk-sharp3.targets" Condition="Exists('..\..\packages\gtk-sharp3\build\gtk-sharp3.targets')" Label="Paket" />
-  <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/OpenTK.GLWidget/OpenTK.GLWidget.csproj]: And here.

File: src/OpenTK/OpenTK.Android.csproj [hunk 5]
```diff
@@ -268,4 +268,12 @@
     <Exec WorkingDirectory="$(SolutionDir)src/Generator.Rewrite/bin/Release" Command="mono ./Rewrite.exe $(SolutionDir)src/OpenTK/bin/Release/Android/OpenTK.dll $(SolutionDir)OpenTK.snk" Condition="$(OS) != 'Windows_NT' and $(Configuration) == 'Release'" />
   </Target>
   <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/OpenTK/OpenTK.Android.csproj]: And here.

File: src/OpenTK/OpenTK.csproj [hunk 6]
```diff
@@ -824,4 +816,12 @@
     </MonoDevelop>
   </ProjectExtensions>
   <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
```
Comment (Nihlus) [inline src/OpenTK/OpenTK.csproj]: And here.

File: src/OpenTK/OpenTK.iOS.csproj [hunk 7]
```diff
@@ -251,4 +251,12 @@
     <Exec WorkingDirectory="$(SolutionDir)src/Generator.Rewrite/bin/Release" Command="mono ./Rewrite.exe $(SolutionDir)src/OpenTK/bin/Release/iOS/OpenTK.dll $(SolutionDir)OpenTK.snk -dllimport" Condition="$(OS) != 'Windows_NT' and $(Configuration) == 'Release'" />
   </Target>
   <Import Project="..\..\build\StyleCop.props" />
+  <ItemGroup>
+    <Analyzer Include="..\..\packages\StyleCop.Analyzers\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll">
```
Comment (Nihlus) [inline src/OpenTK/OpenTK.iOS.csproj]: And here.

File: src/Generator.Bind/Generator.Bind.csproj (hunk 1 above)
Comment (Frassle) [reply src/Generator.Bind/Generator.Bind.csproj]: It is but running paket puts it here as well. Not sure why.

Comment (Nihlus) [reply src/Generator.Bind/Generator.Bind.csproj]: It doesn't resolve the imports, I don't think.

Comment (Frassle) [reply src/Generator.Bind/Generator.Bind.csproj]: This seems to break the StyleCop disabled build as well. Will need to look into this some.

**Comment by Frassle** [comment]
I've removed the analyzers entry from StyleCop.props as paket adds it everywhere every time its run anyway. On msbuild 2017 (at least) this has broken the conditional running of StyleCop because the RunCodeAnalysis doesn't seem to actually change if StyleCop runs or not, we have to remove the Analyzers item somehow and I can't find any way to do that which still plays nice with paket.

**Comment by Nihlus** [comment]
@Frassle What you could do is add a condition block for the project import statement, completely cutting out StyleCop if the relevant env variable is set, that is,

```xml
<Import Project="..\..\build\StyleCop.props" Condition=" '$(SkipStyleCop)' == ''"/>
```

**Comment by Frassle** [comment]
That doesn't work, because paket adds references to the StyleCop analyzers to the main project files. So the only thing really left in StyleCop.props is the path to the rules file. I'm inclined to just clean the whole thing up, remove the whole idea of BuildWithStyleCop and StyleCop.props get this paket stuff sorted and then see about adding the conditional style cop stuff in (maybe with help, because damned if I can work out how to do it cleanly)

**Comment by Nihlus** [comment]
Right. What we could do is just keep an eye on what paket adds manually, but hopefully these issues will go away when we move to the new project format.

**Comment by Frassle** [comment]
I've just removed the StyleCop stuff for now. I couldn't find any way to make it work and I want to get this paket check in, people keep being thrown by project diffs.

**Comment by Nihlus** [comment]
LGTM

**Pull request merged**